# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.cpp
# Commit: 047bcbf3769f
# Full Hash: 047bcbf3769f8dcdeda5f00e8dd1b0cad893e602
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 21:58:52
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 8. Remove unused visibility observer from HTMLCanvasElement. r=jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D130784
# ==============================================================================

diff -r 2859a91f3a73 -r 047bcbf3769f dom/html/HTMLCanvasElement.cpp
--- a/dom/html/HTMLCanvasElement.cpp	Fri Dec 10 02:57:54 2021 +0000
+++ b/dom/html/HTMLCanvasElement.cpp	Fri Dec 10 02:57:54 2021 +0000
@@ -269,7 +269,6 @@
 HTMLCanvasElementObserver::HTMLCanvasElementObserver(
     HTMLCanvasElement* aElement)
     : mElement(aElement) {
-  RegisterVisibilityChangeEvent();
   RegisterObserverEvents();
 }
 
@@ -277,28 +276,9 @@
 
 void HTMLCanvasElementObserver::Destroy() {
   UnregisterObserverEvents();
-  UnregisterVisibilityChangeEvent();
   mElement = nullptr;
 }
 
-void HTMLCanvasElementObserver::RegisterVisibilityChangeEvent() {
-  if (!mElement) {
-    return;
-  }
-
-  Document* document = mElement->OwnerDoc();
-  document->AddSystemEventListener(u"visibilitychange"_ns, this, true, false);
-}
-
-void HTMLCanvasElementObserver::UnregisterVisibilityChangeEvent() {
-  if (!mElement) {
-    return;
-  }
-
-  Document* document = mElement->OwnerDoc();
-  document->RemoveSystemEventListener(u"visibilitychange"_ns, this, true);
-}
-
 void HTMLCanvasElementObserver::RegisterObserverEvents() {
   if (!mElement) {
     return;
@@ -348,19 +328,6 @@
   return NS_OK;
 }
 
-NS_IMETHODIMP
-HTMLCanvasElementObserver::HandleEvent(Event* aEvent) {
-  nsAutoString type;
-  aEvent->GetType(type);
-  if (!mElement || !type.EqualsLiteral("visibilitychange")) {
-    return NS_OK;
-  }
-
-  mElement->OnVisibilityChange();
-
-  return NS_OK;
-}
-
 NS_IMPL_ISUPPORTS(HTMLCanvasElementObserver, nsIObserver)
 
 // ---------------------------------------------------------------------------
@@ -1278,22 +1245,6 @@
   return LayersBackend::LAYERS_NONE;
 }
 
-void HTMLCanvasElement::OnVisibilityChange() {
-  if (OwnerDoc()->Hidden()) {
-    return;
-  }
-
-  if (mOffscreenCanvas) {
-    MOZ_CRASH("todo");
-    // Dispatch to GetActiveEventTarget.
-    return;
-  }
-
-  if (mCurrentContext) {
-    mCurrentContext->OnVisibilityChange();
-  }
-}
-
 void HTMLCanvasElement::OnMemoryPressure() {
   if (mOffscreenCanvas) {
     MOZ_CRASH("todo");